src/expr/impl/src/external/iceberg.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module contains the expression for computing the iceberg partition value.
+//! spec ref: <https://iceberg.apache.org/spec/#partition-transforms>
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+
+use icelake::types::{
+    BoxedTransformFunction, Bucket as BucketTransform, Day as DayTransform, Hour as HourTransform,
+    Month as MonthTransform, Truncate as TruncateTransform, Year as YearTransform,
+};
+use risingwave_common::array::{ArrayRef, DataChunk};
+use risingwave_common::ensure;
+use risingwave_common::row::OwnedRow;
+use risingwave_common::types::{DataType, Datum};
+use risingwave_expr::expr::BoxedExpression;
+use risingwave_expr::{build_function, Result};
+
+#[derive(Debug)]
+pub enum IcebergTransformType {
```

Comment:
I guess defining a new enum is because we need a fixed mapping between transform types and integers?

Reply:
> I guess defining a new enum is because we need a fixed mapping between transform types and integers?

This way maybe a function would be enough?

Reply:
```
impl<'a> ToString for &'a Transform {
    fn to_string(&self) -> String {
        match self {
            Transform::Identity => "identity".to_string(),
            Transform::Year => "year".to_string(),
            Transform::Month => "month".to_string(),
            Transform::Day => "day".to_string(),
            Transform::Hour => "hour".to_string(),
            Transform::Void => "void".to_string(),
            Transform::Bucket(length) => format!("bucket[{}]", length),
            Transform::Truncate(width) => format!("truncate[{}]", width),
        }
    }
}
```

I find that we can use string so that we don't even need extra function meta param( all info contains in the string

Reply:
Yes, it's Just serializaion/deserializatiom